publication-commons/src/main/java/no/unit/nva/publication/model/business/Resource.java

```diff
@@ -219,15 +225,55 @@ public Resource fetch(ResourceService resourceService) throws NotFoundException
         return resourceService.getResourceByIdentifier(this.getIdentifier());
     }
 
-    public void republish(ResourceService resourceService, UserInstance userInstance) {
-        if (!UNPUBLISHED.equals(getStatus())) {
-            throw new IllegalStateException("Only unpublished resource can be republished");
+    public void publish(ResourceService resourceService, UserInstance userInstance) throws NotFoundException {
+        var resource = this.fetch(resourceService);
+        if (resource.isNotPublished()) {
+            resource.publish(userInstance);
+            resourceService.updateResource(resource);
+        }
+    }
+
+    private void publish(UserInstance userInstance) {
+        if (this.isPublished()) {
+            return;
+        }
+        if (this.isNotPublishable()) {
+            throw new IllegalStateException("Resource is not publishable!");
+        } else {
+            this.setStatus(PUBLISHED);
+            var currentTime = Instant.now();
+            this.setPublishedDate(currentTime);
+            this.setResourceEvent(PublishedResourceEvent.create(userInstance, currentTime));
+        }
```

Comment:
nitpick: Perhaps the two `publish`methods could be combined? The logic  seem to just check `isNotPublished` twice, in a roundabout way.

Reply:
Also, I don't think you need to wrap the last bit here in an `else`, since you can't get there if an exception is thrown.